crates/starknet-devnet-server/src/api/json_rpc/mod.rs

```diff
@@ -351,33 +351,26 @@ impl std::fmt::Display for StarknetRequest {
 #[cfg_attr(test, derive(Deserialize))]
 #[serde(untagged)]
 pub enum StarknetResponse {
-    BlockWithTransactionHashes(Block),
-    BlockWithFullTransactions(Block),
-    BlockWithReceipts(Block),
+    Block(Block),
```

Comment:
It's related to this sentence from the PR description:
> StarknetResponse enum variants are refactored to reflect the return type, not the method that the variant corresponds to.

Why is this simplification only introduced now? Was it not possible to do so before?

Reply:
It was possible of course. My idea at the beginning was to match every request variant to a response variant, but by using #[serde(untagged)] it deserialises to the first variant that matches to the response object structure, which often turns out to be different variant than intended. Because multiple RPC methods have the same return type

Reply:
To the question why it is introduced now, its either low priority and now I find time for it